source/language/openpulse.rst

```diff
@@ -265,7 +265,7 @@ samples provides optimization opportunities that wouldn't be available otherwise
    :force:
 
    // arbitrary complex samples
-   waveform arb_waveform = [1+0im, 0+1im, 1/sqrt(2)+1/sqrt(2)im];
+   waveform arb_waveform = {1+0im, 0+1im, 1/sqrt(2)+1/sqrt(2)im};
```

Comment:
This is technically a change to the OpenPulse specification (based purely on the text of this patch) - the OpenPulse `waveform` type isn't inherently linked to the OpenQASM 3 `array` type, so there's no hard requirement that their literals actually need to have the same syntax.  I don't know precisely what state the OpenPulse specification is in, nor what implementors are doing.  I think the `openpulse-python` project is implicitly assuming that the waveform literal is the same as the `array` literal, from the way it inherits from the `qasm3Lexer.g4` tokens, but given this text, that's not actually correct.

Let me throw this over to @openqasm/wg-openpulse to comment on what the right answer here is, and for somebody to shepherd any other changes needed - I suspect the right answer is that the examples _do_ need updating as in this PR.

(On a semi-related note, the complex-number definitions in this example will likely fail to parse, and I don't think there's any explicit support for it in the OpenPulse spec.  Certainly it would be invalid OQ3 since #358, and OpenPulse usually just reuses the OpenQASM 3 rules.)